shared/src/simulation/utils/randomness.ts

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable no-bitwise */
+import {
+    ArrayMaxSize,
+    ArrayMinSize,
+    IsArray,
+    IsInt,
+    Max,
+    Min,
+    ValidateIf,
+} from 'class-validator';
+import { v4 } from 'uuid';
+import { getCreate } from '../../models/utils';
+import type { ExerciseState } from '../../state';
+import type { Mutable, UUID } from '../../utils';
+import { IsLiteralUnion, IsValue } from '../../utils/validators';
+
+// 4 unsigned 32-bit integers
+type Sfc32state = readonly [number, number, number, number];
+
+export class RandomState {
+    @IsValue('randomState' as const)
+    readonly type = 'randomState';
+
+    @IsLiteralUnion({ sfc32: true })
+    readonly algo = 'sfc32';
+
+    @ValidateIf((o) => o.algo === 'sfc32')
+    @IsArray()
+    @ArrayMinSize(4)
+    @ArrayMaxSize(4)
+    @IsInt({ each: true })
+    @Min(0, { each: true })
+    @Max(4294967295 /* UINT32_MAX */, { each: true })
```

Comment:
There can be cases in which elements of the sfc state are bigger than this value which causes subsequent validations (e.g. when ex- and importing an exercise) to fail.